plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/kolo/xmlrpc"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server      string   `toml:"url"`
+	UseIdentTag bool     `toml:"use_identification_tag"`
+	MetricsInc  []string `toml:"metrics_include"`
+	MetricsExc  []string `toml:"metrics_exclude"`
+
+	rpcClient   *xmlrpc.Client
+	fieldFilter filter.Filter
+}
+
+type processInfo struct {
+	Name          string `xmlrpc:"name"`
+	Group         string `xmlrpc:"group"`
+	Description   string `xmlrpc:"description"`
+	Start         int32  `xmlrpc:"start"`
+	Stop          int32  `xmlrpc:"stop"`
+	Now           int32  `xmlrpc:"now"`
+	State         int16  `xmlrpc:"state"`
+	Statename     string `xmlrpc:"statename"`
+	StdoutLogfile string `xmlrpc:"stdout_logfile"`
+	StderrLogfile string `xmlrpc:"stderr_logfile"`
+	SpawnErr      string `xmlrpc:"spawnerr"`
+	ExitStatus    int8   `xmlrpc:"exitstatus"`
+	Pid           int32  `xmlrpc:"pid"`
+}
+
+type supervisorInfo struct {
+	StateCode int8   `xmlrpc:"statecode"`
+	StateName string `xmlrpc:"statename"`
+	Ident     string
+}
+
+const sampleConfig = `
+  ## Url of supervisor's XML-RPC endpoint if basic auth enabled in supervisor http server,
+  ## than you have to add credentials to url (ex. http://login:pass@localhost:9001/RPC2)
+  # url="http://localhost:9001/RPC2"
+  ## Use supervisor identification string as server tag
+  use_identification_tag = false
+  ## With settings below you can manage gathering additional information about processes
+  ## If both of them empty, then all additional information will be collected.
+  ## Currently supported supported additional metrics are: pid, rc
+  metrics_include = []
+  metrics_exclude = ["pid", "rc"]
+`
+
+func (s *Supervisor) Description() string {
+	return "Gather info about processes state, that running under supervisor using its XML-RPC API"
+}
+
+func (s *Supervisor) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Supervisor) Gather(acc telegraf.Accumulator) error {
+	// API call to get information about all running processes
+	var err error
+	var rawProcessData []processInfo
+	err = s.rpcClient.Call("supervisor.getAllProcessInfo", nil, &rawProcessData)
+	if err != nil {
+		return fmt.Errorf("failed to get processes info: %v", err)
+	}
+
+	// API call to get information about instance status
+	var status supervisorInfo
+	err = s.rpcClient.Call("supervisor.getState", nil, &status)
+	if err != nil {
+		return fmt.Errorf("failed to get processes info: %v", err)
+	}
+
+	// API call to get identification string
+	err = s.rpcClient.Call("supervisor.getIdentification", nil, &status.Ident)
+	if err != nil {
+		return fmt.Errorf("failed to get instance identification: %v", err)
+	}
+
+	// Iterating through array of structs with processes info and adding fields to accumulator
+	for _, process := range rawProcessData {
+		processTags, processFields, err := s.parseProcessData(process, status)
+		if err != nil {
+			acc.AddError(err)
+		}
```

Comment:
You probably should use `continue` here to skip the addition of the metric
```suggestion
		if err != nil {
			acc.AddError(err)
			continue
		}
```

Reply:
Yeah, thanks for spotting this!